### PR TITLE
Add version flag and print version in logs

### PR DESCRIPTION
**Pull request by redhatrises**

### CODE REVIEW

**Comment by redhatrises** [comment]
I believe the build failure for the container build CI is to be expected since the files don't exist in the repository yet.

File: Dockerfile [hunk 1]
```diff
@@ -11,6 +11,7 @@ RUN go mod download
 
 # Copy the go source
 COPY main.go main.go
+COPY version/ version/
```
Comment (redhatrises) [inline Dockerfile:14]: LOL.... and this is why you don't do this late at night. :D

**Comment by redhatrises** [comment]
> How this works when we use quay.io as a CD?

Open to suggestions.... can default to "devel" or use a commit at least for main. For handling when there is a new GH tag, might need to switch to a GH action workflow with https://github.com/redhat-actions/push-to-registry.

**Comment by isimluk** [comment]
Also, instead of `--version` command-line argument, I would welcome printing out the version early in the main to the log.

(Unless `--version` is required by certification or other such power structure).

**Comment by redhatrises** [comment]
> > Open to suggestions....
> 
> Here is an idea. We already had to hardcode version in the Makefile
> 
> ```
> VERSION ?= 0.5.0
> ```
> 
> and when you change the version here in the makefile it will populate CSV, bundle, etc, multiple places.
> 
> So what about we modify Makefile to write also the `version.go` that will ensure the same process and same value in the metadata as well as inside the binary.

Took a different route and created a build script that the dockerfile then calls on build. This way we can handle official tag/versions, if needed, certification versions, and CD builds which will have versioning that adds a git tag to reflect it is outside of a release tag.

> Also, instead of `--version` command-line argument, I would welcome printing out the version early in the main to the log.
> 
> (Unless `--version` is required by certification or other such power structure).

When the manager is run without args, the first thing that is printed is the version. The cli flag is useful for allowing for grabbing the version and exiting the manager without having to run the manager itself.

**Review by isimluk — APPROVED** [review]

**Pull request merged**